cmd/lotus-worker/main.go

```diff
@@ -296,6 +296,13 @@ var runCmd = &cli.Command{
 			}
 		}
 
+		// ensure tmpdir exists
+		td := os.TempDir()
+		if err := os.MkdirAll(td, 0755); err != nil {
```

Comment:
If the directory exists but your don't have perms that wouldn't detect it I think.

You might want to use `MkdirTemp("", "lotus")` instead maybe since it's going to try and create a new directory in the TempDir.

Reply:
Perms are basically never an issue, but non-existent temp directories tend to be - this is because in many setups you'll set `TMPDIR=/tmp/something`, and just forget to create those directories. It's also worth noting that people tend to run workers as root, because that's easier than setting up all perms needed for everything to work (cuda, memory locking, thread pinning, etc.)